baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StreamFlatMapOptional.java

```diff
@@ -67,7 +68,13 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
             if (receiver != null) {
                 SuggestedFix.Builder fix = SuggestedFix.builder();
                 String optionalType = SuggestedFixes.qualifyType(state, fix, Optional.class.getCanonicalName());
-                String replacement = ".filter(" + optionalType + "::isPresent).map(" + optionalType + "::get)";
+                Type receiverType = ASTHelpers.getType(receiver);
+                com.sun.tools.javac.util.List<Type> receiverTypeArguments = receiverType.getTypeArguments();
+                com.sun.tools.javac.util.List<Type> optionalTypeArgs = receiverTypeArguments.isEmpty()
+                        ? com.sun.tools.javac.util.List.nil()
+                        : receiverTypeArguments.get(0).getTypeArguments();
+                String type = typeArgument(optionalTypeArgs, state, fix);
+                String replacement = "." + type + "mapMulti(" + optionalType + "::ifPresent)";
```

Comment:
I think we may be able to simplify a bit using `ASTHelpers.getType(tree)`, which gives us `Stream<T>` as opposed to unwrapping `Stream<Optional<T>>`

Reply:
thanks, updated to use ASTHelpers as well as handling `Stream.filter(Optional::isPresent).map(Optional::get)` and `filter(Optional::isPresent).map(Optional::orElseThrow)`.